packages/client/src/HierarchySelection.ts

```diff
@@ -0,0 +1,23 @@
+import { Hash, Hierarchy, Product } from "@escad/core"
+import { getHierarchyPath, HierarchyPath } from "./HierarchyPath"
+
+export interface HierarchySelectionPart {
+  type: "include" | "exclude",
+  path: HierarchyPath,
+}
+
+export type HierarchySelection = HierarchySelectionPart[]
+
+export function resolveHierarchySelection(selection: HierarchySelection, hierarchy: Hierarchy){
+  const selected = new Set<Hash<Product>>()
+  for(const selectionPart of selection) {
+    const subHierarchy = getHierarchyPath(selectionPart.path, hierarchy)
+    if(!subHierarchy?.linkedProducts) continue
```

Comment:
does this silently skip over 'invalid' selections?

Reply:
Yes